StreamChatUITestsAppUITests/Tests/QuotedReply_Tests.swift

```diff
@@ -0,0 +1,254 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import XCTest
+
+final class QuotedReply_Tests: StreamTestCase {
+    
+    let messageCount = 30
+    let parentMessage = "1"
+    let quotedMessage = "quoted reply"
+    
+    override func setUpWithError() throws {
+        try XCTSkipIf(ProcessInfo().operatingSystemVersion.majorVersion == 12,
+                      "Quoted Reply automated tests do not work well on iOS 12")
+        try super.setUpWithError()
+        addTags([.coreFeatures])
+    }
+    
+    override func tearDownWithError() throws {
+        if ProcessInfo().operatingSystemVersion.majorVersion > 12 {
+            try super.tearDownWithError()
+        }
+    }
+    
+    func test_quotedReplyInList_whenUserAddsQuotedReply() {
+        linkToScenario(withId: 1667)
+        
+        let messageCount = 20
+        
+        GIVEN("user opens the channel") {
+            backendRobot.generateChannels(count: 1, messagesCount: 20)
+            userRobot.login().openChannel()
+        }
+        WHEN("user adds a quoted reply to participant message") {
+            userRobot
+                .scrollMessageListUp(times: 4)
+                .replyToMessage(quotedMessage, messageCellIndex: messageCount - 1)
+                .waitForMessageVisibility(at: 0)
+        }
+        THEN("user observes the reply in message list") {
+            userRobot
+                .assertQuotedMessage(replyText: quotedMessage, quotedText: parentMessage)
+                .assertScrollToBottomButton(isVisible: false)
+        }
+        WHEN("user taps on a replied message") {
+            userRobot.tapOnRepliedMessage(parentMessage, at: 0)
+        }
+        THEN("user is scrolled up to the parent message") {
```

Comment:
```suggestion
        GIVEN("user opens the channel") {
            backendRobot.generateChannels(count: 1, messagesCount: 20)
            userRobot.login().openChannel()
        }
        WHEN("user adds a quoted reply to participant message") {
            userRobot
                .scrollMessageListUp(times: 4)
                .replyToMessage(quotedMessage, messageCellIndex: messageCount - 1)
                .waitForMessageVisibility(at: 0)
        }
        THEN("user observes the quote reply in message list") {
            userRobot
                .assertQuotedMessage(replyText: quotedMessage, quotedText: parentMessage)
                .assertScrollToBottomButton(isVisible: false)
        }
        WHEN("user taps on a quoted message") {
            userRobot.tapOnRepliedMessage(parentMessage, at: 0)
        }
        THEN("user is scrolled up to the quoted message") {
```

The parent message is actually for thread replies. It seems we are confusing the two concepts here 🤔 

Thread reply != Quote reply

A thread reply has a parent message. A quote reply has a quoted message id, it does not have a parent message.